src/AnimateText.js

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import SplitText from '@moxy/react-split-text';
+import IntersectionObserver from '@researchgate/react-intersection-observer';
+import 'intersection-observer'; // Polyfill to support older browsers
+
+const AnimateText = ({
+    children,
+    className,
+    separator,
+    withObserver,
+    shouldShow,
```

Comment:
I wonder why we need these two props. Can't we just assume, by default, if `shouldShow` is `false` the text is displayed only when it's within the viewport? When `shouldShow` becomes `true`, the `IntersectionObserver` would become deactivated. 🤔

Reply:
I wouldn't do it because sometimes you might need to control the text with the `shouldShow` prop, but you still want the `IntersectionObserver` active so the animation only happens when the text is within the viewport. Does that make sense?

Reply:
My understanding of how the component works is as follows:
- if `withObserver` is `true`, `shouldShow` does nothing be it `true` or `false`. The start of the animation is controlled by/delegated to the intersection observer.
- if `withObserver` is `false`, the intersection observer is disabled and the developer has to control `shouldShow`. When `shouldShow` changes to `true`, the text begins its animation
- in either cases above, the text is always rendered but the animation only kicks in when triggered by the intersection observer or by `shouldShow`.
It the above correct?

Reply:
You are correct, my bad! 

I guess one way to make it work with only one prop would be to check if the `shouldShow` is defined or not and not have a default. And when it is defined (no matter if `true` or `false`) we don't use the intersection observer, and let the developer control it. 

In the end that is what is happening now, the `shouldShow` prop is irrelevant unless the `withObserver` is set to `false`. Thoughts?

Reply:
🤔 From my understanding, all `shouldShow` does is activate the animation, right? The text is always rendered regardless even if we use the intersection observer. So, when `shouldShow` is `false` or `undefined` the behavior is exactly the same...
If `shouldShow` is irrelevant as you said when `withObserver` is `true`, why not just have one prop to determine whether the animation is controlled by the developer or automatically by the intersection observer?

Reply:
Yes, but we would remove the `withObserver` prop. So the only way to see if the text is being controlled or not would be by checking if the `shouldShow` prop is defined or not - we need it on [this line](https://github.com/moxystudio/react-animate-text/pull/1/files#diff-77a05e19585bf2c1a6d7c5a9529d0a32R68).

Reply:
@jdiogopeixoto 👌 
I think it's better than having a prop which does nothing in one specific case. I'd probably rename the prop to `show` then. It's not suitable to prefix it with `is` either.